azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,406 @@
+	if err != nil {
+		return fmt.Errorf("Error deleting Analysis Services Server %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for deletion of Analysis Services Server %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	return nil
+}
+
+func validateAnalysisServicesServerName(v interface{}, k string) (warnings []string, errors []error) {
+	value := v.(string)
+
+	if !regexp.MustCompile(`^[a-z][0-9a-z]{2,62}$`).Match([]byte(value)) {
+		errors = append(errors, fmt.Errorf("%q must begin with a letter, be lowercase alphanumeric, and be between 3 and 63 characters in length", k))
+	}
+
+	return warnings, errors
+}
+
+func validateQuerypoolConnectionMode() schema.SchemaValidateFunc {
+	connectionModes := make([]string, len(analysisservices.PossibleConnectionModeValues()))
+	for i, v := range analysisservices.PossibleConnectionModeValues() {
+		connectionModes[i] = string(v)
+	}
+
+	return validation.StringInSlice(connectionModes, true)
+}
+
+func expandAnalysisServicesServerProperties(d *schema.ResourceData) *analysisservices.ServerProperties {
+	adminUsers := expandAnalysisServicesServerAdminUsers(d)
+
+	serverProperties := analysisservices.ServerProperties{AsAdministrators: adminUsers}
+
+	serverProperties.IPV4FirewallSettings = expandAnalysisServicesServerFirewallSettings(d)
+
+	if querypoolConnectionMode, ok := d.GetOk("querypool_connection_mode"); ok {
+		serverProperties.QuerypoolConnectionMode = analysisservices.ConnectionMode(querypoolConnectionMode.(string))
+	}
+
+	return &serverProperties
+}
+
+func expandAnalysisServicesServerMutableProperties(d *schema.ResourceData) *analysisservices.ServerMutableProperties {
+	adminUsers := expandAnalysisServicesServerAdminUsers(d)
+
+	serverProperties := analysisservices.ServerMutableProperties{AsAdministrators: adminUsers}
+
+	serverProperties.IPV4FirewallSettings = expandAnalysisServicesServerFirewallSettings(d)
+
+	serverProperties.QuerypoolConnectionMode = analysisservices.ConnectionMode(d.Get("querypool_connection_mode").(string))
+
+	return &serverProperties
+}
+
+func expandAnalysisServicesServerAdminUsers(d *schema.ResourceData) *analysisservices.ServerAdministrators {
+	adminUsers := d.Get("admin_users").(*schema.Set)
+	members := make([]string, 0)
+
+	for _, admin := range adminUsers.List() {
+		if adm, ok := admin.(string); ok {
+			members = append(members, adm)
+		}
+	}
+
+	return &analysisservices.ServerAdministrators{Members: &members}
+}
+
+func expandAnalysisServicesServerFirewallSettings(d *schema.ResourceData) *analysisservices.IPv4FirewallSettings {
+	firewallSettings := analysisservices.IPv4FirewallSettings{}
+
+	if enablePowerBi, exists := d.GetOkExists("enable_power_bi_service"); exists {
+		firewallSettings.EnablePowerBIService = utils.Bool(enablePowerBi.(bool))
+	} else {
+		firewallSettings.EnablePowerBIService = utils.Bool(false)
+	}
```

Comment:
since the default value for this will be `false` we can just default this above:

```suggestion
	firewallSettings := analysisservices.IPv4FirewallSettings{
		EnablePowerBIService: utils.Bool(d.Get("enable_power_bi_service").(bool))
	}
```